keyboards/gggw/crosses/crosses.c

```diff
@@ -0,0 +1,227 @@
+    uint16_t current   = config->mse_cpi;
+    uint16_t requested = current += inc ? 50 : -50;
+
+    if (requested < MIN_DEFAULT_DPI) {
+#ifdef CONSOLE_ENABLE
+        dprintf("%u is beyond bounds - omitting operation", requested);
+#endif // CONSOLE_ENABLE
+        return;
+    }
+
+    if (requested > MAX_DEFAULT_DPI) {
+#ifdef CONSOLE_ENABLE
+        dprintf("%u is beyond bounds - omitting operation", requested);
+#endif // CONSOLE_ENABLE
+        return;
+    }
+
+    config->mse_cpi = requested;
+
+#ifdef POINTING_DEVICE_COMBINED
+    pointing_device_set_cpi_on_side(false, get_pointer_dpi(config));
+#endif /* ifdef POINTING_DEVICE_COMBINED */
+
+#if defined(POINTING_DEVICE_LEFT) || defined(POINTING_DEVICE_RIGHT)
+    pointing_device_set_cpi(get_pointer_dpi(config));
+#endif /* ifdef POINTING_DEVICE_* */
+
+    write_config_to_eeprom(&global_user_config);
+}
+
+void debug_config_to_console(global_user_config_t* config) {
+#ifdef CONSOLE_ENABLE
+    dprintf("(crosses) process_record_user: config = {\n"
+            "\traw = 0x%lu,\n"
+            "\t{\n"
+            "\t\tmse_cpi=0x%X (%u)\n"
+            "\t}\n"
+            "}\n",
+            (unsigned long)config->raw, config->mse_cpi, get_pointer_dpi(config));
+#endif // CONSOLE_ENABLE
+}
+
+void eeconfig_init_user(void) {
+    global_user_config.raw     = 0;
+    global_user_config.mse_cpi = MIN_DEFAULT_DPI;
+
+    write_config_to_eeprom(&global_user_config);
+    debug_config_to_console(&global_user_config);
+}
+
+/***********************************************************************
+ * Single Trackball Config
+ ***********************************************************************/
+
+#if defined(POINTING_DEVICE_RIGHT) || defined(POINTING_DEVICE_LEFT)
+
+report_mouse_t pointing_device_task_user(report_mouse_t mouse_report) {
+    if (set_scrolling) {
+        // Calculate and accumulate scroll values based on mouse movement and divisors
+        scroll_acc_h += (float)mouse_report.x / SCROLL_DIVISOR_H;
+        scroll_acc_v += (float)mouse_report.y / SCROLL_DIVISOR_V;
+
+        // Assign integer parts of accumulated scroll values to the mouse report
+        mouse_report.h = (int16_t)scroll_acc_h;
+        mouse_report.v = (int16_t)scroll_acc_v;
+
+        // Update accumulated scroll values by subtracting the integer parts
+        scroll_acc_h -= (int16_t)scroll_acc_h;
+        scroll_acc_v -= (int16_t)scroll_acc_v;
+
+        // Clear the X and Y values of the mouse report
+        mouse_report.x = 0;
+        mouse_report.y = 0;
+    }
+
+    return mouse_report;
+}
+
+#endif /* RIGHT || LEFT */
+
+/***********************************************************************
+ * Dual Trackball Config
+ ***********************************************************************/
+
+#ifdef POINTING_DEVICE_COMBINED
+
+report_mouse_t pointing_device_task_combined_user(report_mouse_t left_report, report_mouse_t right_report) {
```

Comment:
This should also use the `_kb` function, and call the `_user` function appropriately. Might need tweaking a bit with the return value.